File: 013.RomanToInteger/wyrj/RomanToInteger.c
/* 28ms */
int romanToInt(char* s) {
    int ret = 0;
    int index = 0, i, len;
    int translator[] = {1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1};
    char *translatee[] = {"M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"};
    for (i = 0; i < 13; i++) {
        len = strlen(translatee[i]);
        while(index < strlen(s) && 0 == strncmp(&s[index], translatee[i], len)) {
            index += len;
            ret += translator[i];
        }
    }
    return ret;
}

File: 013.RomanToInteger/wyrj/RomanToInteger.js
/* 356ms */
/**
 * @param {string} s
 * @return {number}
 */
var romanToInt = function(s) {
    var translator = [[1000, 'M'], [900, 'CM'],
                    [500, 'D'], [400, 'CD'],
                    [100, 'C'], [90, 'XC'],
                    [50, 'L'], [40, 'XL'], 
                    [10, 'X'], [9, 'IX'], 
                    [5, 'V'], [4, 'IV'], [1, 'I']];
    var ret = 0;
    for (var i = 0; i < translator.length; i++) {
        while(0 === s.indexOf(translator[i][1])) {
            ret += translator[i][0];
            s = s.slice(translator[i][1].length);
        }
    }
    return ret;
};